### DISCUSSION TITLE
A note on encoding

### DISCUSSION THREAD

**Discussion by vincent-lg** [Blightmud/Blightmud#279] [Q&A] [answered]
Hi,
This is not explicitly mentioned in the documentation I checked, but does BlightMUD support other encodings than ASCII (and utf-8)?  It seems so, from my brief attempts with a MUD I know was running using `latin-1`, but it might be a catch on the server and the client itself.  In any case this is a good news for users who aren't playing ASCII-only MUDs.
These users are usually concerned with encoding matching on the client and server.  Usually, and as far as I know, there is no Telnet protocol to exchange the encoding between server and client, though there is possibly one to set utf-8 to on or off (good... if you use utf-8, which of course you should, but not everyone does).  The next thing, of course, is a client setting to change encoding (output encoding and input encoding tend to be similar for most users, at least, I never came across a reason to have different ones).
And after that comes the great question: does BlightMUD support multi-byte characters?  In utf-8, `é` is split on two bytes for instance.  While `TT++` seemed to accept that, when writing `é` then backspace, only one of these bytes were removed, which sent gibberish to the server (which usually decided it wasn't utf-8 and reverted to a more permissive encoding).  Again I didn't notice this error on BlightMUD, but I didn't notice specific documentation on this topic either.
Again, thanks for this great work!

**Comment by LiquidityC** [comment, marked as answer]
Hi @vincent-lg,

I'll try to answer your questions as well as I can. Let's just say that I'm no expert on encodings and mainly just stick to utf-8 with the knowledge that it throws a wide net and can play nice with the earlier encodings.

We are pretty locked down with utf-8 for the time being. Mainly due to it being standard in Rust. As far as i know utf-8 will play nice with basic ascii and latin-1 and a load of other encodings so there should be no valid reason to switch "down" to those as far as i know. Then again I'm no expert on the intricacies of char encodings.

We do have a small russian playerbase which comes with a lot of advanced multi byte characters. As far as i know it's working fine for them at the moment. So as long as a server sends correct utf-8 we can deal with it in all shapes and sizes.

So the short answer is. It's utf-8 all the way and should work fine with anything that is a "subset" of utf-8 (and this is a long list). As for switching. That is a heavy question where I only dare say that it might require some work. And if it's not switching to a higher level of encoding utf-16 etc. it might not take a big priority.

I hope that answers all your questions in a acceptable way.

Br,
Linus
